pkg/apis/pipeline/v1alpha1/condition_types.go

```diff
@@ -0,0 +1,82 @@
+/*
+ *
+ * Copyright 2019 The Tekton Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * /
+ */
+
+package v1alpha1
+
+import (
+	"github.com/knative/pkg/apis"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// Check that Task may be validated and defaulted.
+var _ apis.Validatable = (*Condition)(nil)
+
+// +genclient
+// +genclient:noStatus
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// Condition declares a step that is used to gate the execution of a Task in a Pipeline.
+// A condition execution (ConditionCheck) evaluates to either true or false
+// +k8s:openapi-gen=true
+type Condition struct {
+	metav1.TypeMeta `json:",inline"`
+	// +optional
+	metav1.ObjectMeta `json:"metadata"`
+
+	// Spec holds the desired state of the Condition from the client
+	// +optional
+	Spec ConditionSpec `json:"spec"`
+}
+
+// ConditionSpec defines the desired state of the Condition
+type ConditionSpec struct {
+	// Check declares container whose exit code determines where a condition is true or false
+	Check corev1.Container `json:"check,omitempty"`
+
+	// Params is an optional set of parameters which must be supplied by the user when a Condition
+	// is evaluated
+	// +optional
+	Params []ParamSpec `json:"params,omitempty"`
+}
+
+
+// ConditionCheck represents a single evaluation of a Condition step.
+type ConditionCheck TaskRun
+
+// ConditionCheckStatus is the observed state of a ConditionCheck
+type ConditionCheckStatus TaskRunStatus
```

Comment:
I think you said you're already digging a bit into refactoring this so that we don't reuse Task runs but here's some more food for thought! Looking at what we've got in `TaskRunStatus` we've got these fields:

```
    Results *Results "json:\"results,omitempty\"" # Oh boy what even is this? Should probably delete this... 😅 

    # These seems like they make sense for a Condition
    Status "json:\",inline\""
    PodName string "json:\"podName\""
    StartTime *Time "json:\"startTime,omitempty\""
    CompletionTime *Time "json:\"completionTime,omitempty\""

    # Since we only have one step ever it's kinda odd that we have a "steps" field with multiple steps in it
    Steps []StepState "json:\"steps,omitempty\""

    # I'm doubtful these make sense for a condition
    RetriesStatus []TaskRunStatus "json:\"retriesStatus,omitempty\""
    ResourcesResult []PipelineResourceResult "json:\"resourcesResult,omitempty\""

   # Would it make sense to have something explicit like "condition result" to say "pass" or "fail" or similar?
   # (Just ignore if it's not worth it)
```

My inclination here would be to add as little functionality as we can possibly get away with - it's always easier to add more later than to remove it!

Reply:
Totally agree. I think reusing the `TaskRunStatus` ties us to the TaskRun implementation which is not something we want!. Will have this updated soon!

Reply:
Removed the Results, RetriesResults, ResourceResult, and Steps fields. Added a `Check` field that is of type `ContainerState`.

Also, from the comment for [Results](https://github.com/tektoncd/pipeline/blob/master/pkg/apis/pipeline/v1alpha1/taskrun_types.go#L118), it seems like it points to `location logs have been uploaded to` though it seems like its not fully implemented (cc @sbwsg )

Reply:
> Also, from the comment for Results, it seems like it points to location logs have been uploaded to though it seems like its not fully implemented (cc @sbwsg )

No worries leaving that as-is - almost definitely leftover from our first attempt at the API 😅